diode-server/netboxdiodeplugin/client.go

```diff
@@ -96,7 +101,24 @@ type Client struct {
 
 // NewClient creates a new NetBox Diode plugin client
 func NewClient(logger *slog.Logger, apiKey string) (*Client, error) {
-	rt, err := newAPIRoundTripper(apiKey, http.DefaultTransport)
+
+	transport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: skipTLS(),
+		},
+	}
```

Comment:
To have it tested in more clear fashion, could we extract it into a function, i.e. `NewHTTPTransport()` and have a dedicated test asserting something like this?

```go
transport := netboxdiodeplugin.NewHTTPTransport()
assert.Equal(t, tt.expectedInsecure, transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify)
```

Reply:
done!